server/src/test/java/com/vaadin/data/BinderTest.java

```diff
@@ -494,4 +494,39 @@ public void setReadOnly_binderLoadedByReadBean() {
         Assert.assertFalse(ageField.isReadOnly());
     }
 
+    @Test
+    public void isValidTest() {
+        binder.forField(nameField)
+                .withValidator(
+                        Validator.from(
+                                name -> !name.equals("fail field validation"),
+                                ""))
+                .bind(Person::getFirstName, Person::setFirstName);
+
+        binder.withValidator(
+                Validator.from(person -> !person.getFirstName()
+                        .equals("fail bean validation"), ""));
+
+        binder.setBean(item);
+
+        Assert.assertTrue(binder.isValid());
+
+        nameField.setValue("fail field validation");
+        Assert.assertFalse(binder.isValid());
+
+        nameField.setValue("");
+        Assert.assertTrue(binder.isValid());
+
+        nameField.setValue("fail bean validation");
+        Assert.assertFalse(binder.isValid());
+
+        // removing the bean should make isValid return true again, since bean
```

Comment:
Even though the bean is removed, technically the value in would still be "invalid". I don't think we need to test that it returns explicitly true, since it's more of "undefined" than "true"